src/Engine/Image.php

```diff
@@ -0,0 +1,88 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\Engine;
+
+use Exception;
+use Imagine\Filter\Basic\Crop;
+use Imagine\Image\Box;
+use Imagine\Image\Point;
+
+class Image
+{
+
+    /** @var string Original URL of image. */
+    private $imageUrl;
+
+    /** @var int[] Array of floats with keys: `x`, `y`, `width`, and `height`. */
+    private $crop;
+
+    /** @var string|null Image data. */
+    private $data;
+
+    /** @var int|null Image size in bytes. */
+    private $size;
+
+    /**
+     * @param int[] $crop
+     */
+    public function __construct(string $imageUrl, array $crop = [])
+    {
+        $this->imageUrl = $imageUrl;
+        $this->crop = $crop;
+    }
+
+    public function getUrl(): string
+    {
+        return $this->imageUrl;
+    }
+
+    public function needsCropping(): bool
+    {
+        return isset($this->crop['width']) && $this->crop['width'] > 0
+            && isset($this->crop['height']) && $this->crop['height'] > 0;
+    }
+
+    public function getCrop(): Crop
+    {
+        return new Crop(
+            new Point($this->crop['x'], $this->crop['y']),
+            new Box($this->crop['width'], $this->crop['height'])
+        );
+    }
+
+    public function hasData(): bool
+    {
+        return null !== $this->data;
+    }
+
+    public function getData(): string
+    {
+        if (null === $this->data) {
+            throw new Exception('Image::setData() must be called before getData()');
```

Comment:
Suggest LogicException or RuntimeException for this and line 79

Reply:
Done.